### ISSUE TITLE
Installation issue: flux-core: Concretization error due to libxml2 variant

### ISSUE THREAD

**Issue by SteVwonder** [spack/spack#14827]
### Steps to reproduce the issue

```console
$ spack install flux-core@master
==> Error: An unsatisfiable variant constraint has been detected for spec:

    gettext@0.20.1%gcc@7.4.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu18.04-broadwell


while trying to concretize the partial spec:

    python@3.7.6%gcc@7.4.0+bz2+ctypes+dbm~debug~libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4~uuid+zlib arch=linux-ubuntu18.04-broadwell
        ^pkgconf@1.6.3%gcc@7.4.0 arch=linux-ubuntu18.04-broadwell


python requires gettext variant ~libxml2, but spec asked for +libxml2
```

### Platform and user environment

I am using Github Actions with `ubuntu-latest` as the OS (ubuntu 18.04)
```commandline
$ spack get compilers
compilers:
- compiler:
    spec: clang@6.0.1
    paths:
      cc: /usr/bin/clang-6.0
      cxx: /usr/bin/clang++-6.0
      f77:
      fc:
    flags: {}
    operating_system: ubuntu18.04
    target: x86_64
    modules: []
    environment:
      unset: []
    extra_rpaths: []
- compiler:
    spec: gcc@7.4.0
    paths:
      cc: /usr/bin/gcc-7
      cxx: /usr/bin/g++-7
      f77:
      fc:
    flags: {}
    operating_system: ubuntu18.04
    target: x86_64
    modules: []
    environment: {}
    extra_rpaths: []
``` 

You can view the full workflow log from the github action here: https://github.com/flux-framework/flux-spack/runs/431181960?check_suite_focus=true

### Additional information

This doesn't seem to be an issue on my mac, so I wonder if it is somehow related to the OS or the architecture (broadwell).

Also worth noting that `spack install flux-core@master ^gettext~libxml2` works just fine.  If necessary, I am happy to add a hint/conflict in the `flux-core` package to make this work.  But that seems a bit heavy handed.

**Comment by scheibelp** [comment]
I'm having trouble replicating this on a docker 18.04 Ubuntu image with GCC 7.4.0 - do you have anything in your packages.yaml?

**Comment by SteVwonder** [comment]
Thanks for taking a look!

> do you have anything in your packages.yaml?

Yeah:

```
packages:
  openssl:
    buildable: False
    paths:
      openssl@1.1.1: /usr
  cmake:
    buildable: False
    paths:
      cmake@3.12.4: /usr
  perl:
    buildable: False
    paths:
      perl@5.26.1: /usr
  findutils:
    buildable: False
    paths:
      findutils@4.6.0: /usr
```

Source: https://github.com/flux-framework/flux-spack/blob/master/.ci/packages.yaml

`config.yaml` is empty other than to turn ccache on.

Everything else is "stock spack", done in a clean VM.

**Comment by scheibelp** [comment]
Thanks for the info!

Additional question: what does `spack --version` say?

**Comment by adamjstewart** [comment]
This should be fixed by #14795, can you confirm?

**Comment by SteVwonder** [comment]
> This should be fixed by #14795, can you confirm?

Yep!  Can confirm that it was fixed by that PR.  Thanks!
